barustenberg/src/transcript.rs

```diff
@@ -428,10 +429,10 @@ impl<H: BarretenHasher, Fr: Field, G1Affine: AffineRepr> Transcript<H, Fr, G1Aff
         &self,
         challenge_name: &str,
         idx: usize,
-    ) -> &GenericArray<u8, H::PrngOutputSize> {
+    ) -> Result<&GenericArray<u8, H::PrngOutputSize>, Error> {
         info!("get_challenge(): {}", challenge_name);
         assert!(self.challenges.contains_key(challenge_name));
-        &self.challenges.get(challenge_name).unwrap()[idx].data
+        Ok(&self.challenges.get(challenge_name).unwrap()[idx].data)
```

Comment:
ok around an unwrap?

Reply:
It's because this now returns a result (Better, but not great yet) but the get challenge has specific error types.